### ISSUE TITLE
Document IReadDescriptor and IWriteDescriptor

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3570]
|<img alt="jparise's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| jparise reported|
|-|-|
|Trac ID|trac#3570|
|Type|defect|
|Created|2008-12-10 06:14:38Z|

IReadDescriptor and IWriteDescriptor don't have interface-level documentation.  The attached patch attempts to correct that.

Attachments:

* [interfaces.docs.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/798/798a912d82f4b01e96a6489663985c6052e89de3/4ef44fbdef136136cc6ea8eefbd1ed8f3141d075.diff) (958 bytes) - added by jparise on 2008-12-10 06:15:18Z - IReadDescriptor and IWriteDescriptor documentation
* [interfaces.docs2.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/798/798a912d82f4b01e96a6489663985c6052e89de3/8852a14826feb3587a3ef5f7b28de67d6f86988b.diff) (1915 bytes) - added by jparise on 2008-12-11 06:58:06Z - Improved docs for IFileDescriptor and friends

<details><summary>Searchable metadata</summary>

```
trac-id__3570 3570
type__defect defect
reporter__jparise jparise
priority__low low
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__documentation documentation
time__1228889678000000 1228889678000000
changetime__1229034195000000 1229034195000000
version__None None
owner__ 
cc__thijs
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun set owner to jparise|
|-|-|

Thanks.

The existing documentation for `IReadWriteDescriptor` which you modeled the new docs on wasn't the best.  The interface is abstract.  It's not really supposed to refer to a particular specific implementation.  What would be good is if these docstrings followed the example of `IFileDescriptor` instead, although they could also give a bit more depth than just _file descriptor_.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_3"></a>|@thijstriemstra commented|
|-|-|

Is this a duplicate of [#11358](https://github.com/twisted/twisted/issues/11358)?

**Comment by twisted-trac** [comment]
|<img alt="jparise's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_4"></a>|jparise commented|
|-|-|

Replying to [thijs](#note_3):
> Is this a duplicate of [#11358](https://github.com/twisted/twisted/issues/11358)?

I think they are certainly related, but [#11358](https://github.com/twisted/twisted/issues/11358) is more about the programmatic interface and less about how it's documented.

**Comment by twisted-trac** [comment]
|<img alt="jparise's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_5"></a>|jparise commented|
|-|-|

Thanks for the review.  I thought that existing documentation looked a little off. and I shouldn't have been so quick to base new documentation on it.

I've made another pass.  Additional feedback is welcome.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun set owner to @exarkun|
|-|-|

Looks great to me.  Thanks.  I'll apply the patch.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_7"></a>|@exarkun set status to `closed`|
|-|-|

(In [25636]) Apply interfaces.docs2.diff - improves `I*Descriptor` docstrings

Author: jparise
Reviewer: exarkun
Fixes: [#3570](https://github.com/twisted/twisted/issues/3570)

Add class docstrings for `IReadDescriptor` and `IWriteDescriptor` and remove the references
to a particular concrete implementation of some of these interfaces from the interfaces
themselves (the generated API documentation creates this information automatically).

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
